cukinia

```diff
@@ -363,16 +363,45 @@ _cukinia_process()
 {
 	local pname="$1"
 	local puser="$2"
-	local ret
+	local proc_name
+	local proc_uid
 
 	_cukinia_prepare "Checking process \"$pname\" as ${puser:-any user}"
-	ret=$(ps -a -o user,comm |
-			awk '($2 == "'$pname'") && ($1 ~ /^'${puser:-.*}'$/) {
-			    print;
-			}')
-	[ -n "$ret" ] && return 0 || return 1
-}
 
+	if [ ! -z $puser ]; then
```

Comment:
1. You should double quote `$puser` variable.

Shell may report an error if the variable is left unquoted.

``` zsh
$ if [ -z $puser ]; then echo then; else echo else; fi
zsh: parse error near `else'
```

``` bash
$ if [ -z $puser ]; then echo then; else echo else; fi
bash: syntax error near unexpected token `else'
```

2. You can use `-n` instead of `!` and `-z`.